pkg/controller/master/storagenetwork/storage_network.go

```diff
@@ -0,0 +1,518 @@
+
+	return nil, nil
+}
+
+// return true as finished
+func (h *Handler) checkLonghornSetting(setting *harvesterv1.Setting) (bool, error) {
+	value, err := h.getLonghornStorageNetwork()
+	if err != nil {
+		logrus.Warnf("get Longhorn settings error: %v", err)
+		return false, err
+	}
+
+	if setting.Value == value {
+		// check if we need to restart monitoring pods
+		if !h.checkPodStatusAndStart() {
+			if _, err := h.setConfiguredCondition(setting, false, ReasonInProgress, MsgRestartPod); err != nil {
+				logrus.Warnf("update status error %v", err)
+			}
+			h.settingsController.EnqueueAfter(setting.Name, 5*time.Second)
+			return true, nil
+		}
+
+		// finish
+		if _, err := h.setConfiguredCondition(setting, true, ReasonCompleted, ""); err != nil {
+			logrus.Warnf("update status error %v", err)
+		}
+		return true, nil
+	}
+	return false, nil
+}
+
+// true: all detach
+func (h *Handler) checkLonghornVolumeDetach() (bool, error) {
+	volumes, err := h.longhornVolumeCache.List(util.LonghornSystemNamespaceName, labels.Everything())
+	if err != nil {
+		logrus.Warnf("volume error %v", err)
+		return false, err
+	}
+
+	for _, volume := range volumes {
+		logrus.Infof("volume state: %v", volume.Status.State)
+		if volume.Status.State != "detached" {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func (h *Handler) checkPrometheusStatusAndStart() bool {
+	// check prometheus cattle-monitoring-system/rancher-monitoring-prometheus replica
+	prometheus, err := h.prometheusCache.Get(CattleMonitoringSystemNamespace, RancherMonitoringPrometheus)
+	if err != nil {
+		logrus.Warnf("prometheus get error %v", err)
+		return false
+	}
+
+	logrus.Infof("prometheus: %v", *prometheus.Spec.Replicas)
+	// check started or not
+	if *prometheus.Spec.Replicas == 0 {
+		logrus.Infof("start prometheus")
+		prometheusCopy := prometheus.DeepCopy()
+		if replicas, err := strconv.Atoi(prometheus.Annotations[ReplicaStorageNetworkAnnotation]); err == nil {
+			*prometheusCopy.Spec.Replicas = int32(replicas)
+		}
+		delete(prometheusCopy.Annotations, ReplicaStorageNetworkAnnotation)
+
+		if _, err := h.prometheus.Update(prometheusCopy); err != nil {
+			logrus.Warnf("prometheus update error %v", err)
+			return false
+		}
+		return false
+	}
+
+	return true
+}
+
+func (h *Handler) checkAltermanagerStatusAndStart() bool {
+	// check alertmanager cattle-monitoring-system/rancher-monitoring-alertmanager replica
+	alertmanager, err := h.alertmanagerCache.Get(CattleMonitoringSystemNamespace, RancherMonitoringAlertmanager)
+	if err != nil {
+		logrus.Warnf("alertmanager get error %v", err)
+		return false
+	}
+
+	logrus.Infof("alertmanager: %v", *alertmanager.Spec.Replicas)
+	// check started or not
+	if *alertmanager.Spec.Replicas == 0 {
+		logrus.Infof("start alertmanager")
+		alertmanagerCopy := alertmanager.DeepCopy()
+		if replicas, err := strconv.Atoi(alertmanager.Annotations[ReplicaStorageNetworkAnnotation]); err == nil {
```

Comment:
It looks like we don't handle `err != nil` here.

Reply:
I set default value to handle and ignore error